client/gutenberg/editor/calypsoify-iframe.tsx

```diff
@@ -252,14 +298,20 @@ class CalypsoifyIframe extends Component {
 		this.iframePort.postMessage( { action: 'updateImageBlocks', payload } );
 	};
 
-	openPreviewModal = ( { postUrl, previewPort } ) => {
+	openPreviewModal = ( {
+		postUrl,
+		previewPort,
+	}: {
+		postUrl: string;
+		previewPort: MessagePort;
+	} ) => {
```

Comment:
The `postUrl` and `previewPort` are not really meant as fields in a record type, but rather named/optional parameters of the `openPreviewModal` method. Does TypeScript have some nice way how to express that? I.e., without writing `postUrl` twice?

Reply:
to my knowledge it doesn't make an opinion on this. its our choice to wrap them in an object where it gets weird. the syntax for annotating types is the same syntax as for destructuring renames

```ts
( { postUrl: string } ) => ()
```

In this example are we trying to rename `postUrl` to `string` or declare a string type? the solution is to type the object itself separately as is done above.

---

> not really meant as fields in a record type

The types here aren't trying to convey that they are such. TypeScript is _structural_ and not _nominative_ so all we're trying to convey is that "this function needs an argument that's an object and it needs these things in it."

The way to remove the verbosity is to eliminate the _arguments as an option bag_ approach (which is something I'm all for), though in all fairness, if I could type something twice and eliminate an unexpected bug in my software while shaving off time for other developers who are wanting to know what my function needs, I'll take that trade.

In the meantime, I'll remove the record syntax; these are required parameters anyway. Without types the record syntax can be a nice way of adding some sense of types, though at the cost of runtime allocations.

Reply:
Replaced in f84adf0486

<img width="684" alt="Screen Shot 2019-04-01 at 10 55 12 AM" src="https://user-images.githubusercontent.com/5431237/55348738-c03dfd80-546c-11e9-875f-2915ec926bc1.png">

No information in my IDE was lost by changing `( { postUrl, previewPort } )` into `( postUrl, previewPort )`